src/validation.cpp

```diff
@@ -2233,7 +2233,17 @@ static int64_t nTimePostConnect = 0;
  * part of a single ActivateBestChainStep call.
  */
 struct ConnectTrace {
+private:
     std::vector<std::pair<CBlockIndex*, std::shared_ptr<const CBlock> > > blocksConnected;
+
+public:
+    void BlockConnected(CBlockIndex* pindex, const std::shared_ptr<const CBlock>& pblock) {
+        blocksConnected.emplace_back(pindex, pblock);
+    }
+
+    std::vector<std::pair<CBlockIndex*, std::shared_ptr<const CBlock> > > GetBlocksConnected() {
```

Comment:
Oops, indeed. Now it just returns a reference, which should also be fine. The method is not technically const later (without making things mutable), so I'll leave the second part.